.github/workflows/deploy.yml

```diff
@@ -0,0 +1,33 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.17.1'
+      
+      - name: Install and Build
+        run: |
+          yarn install --frozen-lockfile
+          yarn test
+          yarn build
```

Comment:
could we have this action depend on the build action to remove the duplication here? it looks like we could use the [`needs` parameter](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds). all examples i'm seeing have the 2 jobs in the same file, so i'm not sure if we'll run into issues having them in separate files

Reply:
Nice find! This duplication did not feel quite right. I will see if I can get this working with `needs`. 

If we do have to keep the jobs together in a single file, that might affect where we land on the naming convention (per the earlier comment).

Reply:
Turns out `needs` does need to run within the same workflow (file). I could have created a separate job in the same file and used `needs` in there, but the artifacts from the `build` job would not have been available to it. There are ways to zip up and save artifacts and then pull them down in another job, but it felt like overkill for this use case so I added a simple `if` condition after the build step. If we want to do a more complex workflow in the future we can always break it out to a separate job.